### PR TITLE
Add publisher to ActionAlias

### PR DESCRIPTION
**Pull request by ubaumann**
Add publisher to ActionAlias to get events for CUD Action Aliases.
This change enables DroidStorm (and other systems) to subscribe the events and get informed when a new Alias is created, updated or deleted.

Manual tested:
- Start st2 (`./tools/launchdev.sh startclean`)
- Start queue_consumer
- Create and delete action-alias

Created and deleted action-alias
```bash
(virtualenv) ins@ubuntu:~/st2$ st2 action-alias create contrib/hello_st2/aliases/alias1.yaml
+----------------------+------------------------------+
| Property             | Value                        |
+----------------------+------------------------------+
| id                   | 633f3ceb2735066974631da4     |
| name                 | greet                        |
| pack                 | hello_st2                    |
| description          | Greet StackStorm             |
| ack                  |                              |
| action_ref           | hello_st2.greet              |
| enabled              | True                         |
| extra                |                              |
| formats              | [                            |
|                      |     "greet {{greeting}}"     |
|                      | ]                            |
| immutable_parameters |                              |
| metadata_file        |                              |
| ref                  | hello_st2.greet              |
| result               |                              |
| uid                  | action_alias:hello_st2:greet |
+----------------------+------------------------------+
(virtualenv) ins@ubuntu:~/st2$ st2 action-alias delete greet
Resource with id "greet" has been successfully deleted.
```

Queu consumer
```bash
(virtualenv) ins@ubuntu:~/st2$ ST2_CONFIG_PATH=conf/st2.dev.conf python tools/queue_consumer.py --exchange st2.actionalias
===================================================
Received message
message.properties:
{'application_headers': {},
 'content_encoding': 'binary',
 'content_type': 'application/x-python-serialize',
 'delivery_mode': 2,
 'priority': 0}
message.delivery_info:
{'consumer_tag': 'None1',
 'delivery_tag': 1,
 'exchange': 'st2.actionalias',
 'redelivered': False,
 'routing_key': 'create'}
body:
<ActionAliasDB: ActionAliasDB(ack={}, action_ref="hello_st2.greet", description="Greet StackStorm", enabled=True, extra={}, formats=['greet {{greeting}}'], id=633f3ceb2735066974631da4, immutable_parameters={}, metadata_file="", name="greet", pack="hello_st2", ref="hello_st2.greet", result={}, uid="action_alias:hello_st2:greet")>
===================================================
===================================================
Received message
message.properties:
{'application_headers': {},
 'content_encoding': 'binary',
 'content_type': 'application/x-python-serialize',
 'delivery_mode': 2,
 'priority': 0}
message.delivery_info:
{'consumer_tag': 'None1',
 'delivery_tag': 2,
 'exchange': 'st2.actionalias',
 'redelivered': False,
 'routing_key': 'delete'}
body:
<ActionAliasDB: ActionAliasDB(ack={}, action_ref="hello_st2.greet", description="Greet StackStorm", enabled=True, extra={}, formats=['greet {{greeting}}'], id=633f3ceb2735066974631da4, immutable_parameters={}, metadata_file="", name="greet", pack="hello_st2", ref="hello_st2.greet", result={}, uid="action_alias:hello_st2:greet")>
===================================================
```

Unitest are covering already testing the generic functions. There are no test testing the object specific events. Should I add some test?

Looking forward to the review and please let me know how I can improve the PR.

Pinging @cognifloyd because we talked already in Slack about the PR.

### CODE REVIEW

File: st2common/st2common/transport/actionalias.py
Comment (cognifloyd) [inline st2common/st2common/transport/actionalias.py:2]:
```suggestion
# Copyright 2022 The StackStorm Authors.
```

New file = new copyright line

File: CHANGELOG.rst
Comment (cognifloyd) [inline CHANGELOG.rst:59]:
```suggestion
* Added publisher to ActionAlias to enable streaming ActionAlias create/update/delete events.
```

**Review by cognifloyd — CHANGES_REQUESTED** [review]
Awesome! I just found one more file we need to edit to make these events available in the event stream.

Would you please add the ActionAlias queue to this list:
https://github.com/StackStorm/st2/blob/8496bb2407b969f0937431992172b98b545f6756/st2common/st2common/stream/listener.py#L203-L224

Also, just a couple nitpicks.

Thank you so much for working on this!

We might have to wait a bit before merging this as I'm not sure if it can go in the 3.8 release.

**Comment by ubaumann** [comment]
> Awesome! I just found one more file we need to edit to make these events available in the event stream.
> 
> Would you please add the ActionAlias queue to this list:
> 
> https://github.com/StackStorm/st2/blob/8496bb2407b969f0937431992172b98b545f6756/st2common/st2common/stream/listener.py#L203-L224
>

Thank you very much for finding this place. I totally missed it.


> Also, just a couple nitpicks.

Thank you for reviewing the PR


> We might have to wait a bit before merging this as I'm not sure if it can go in the 3.8 release.

That's totally fine. I was lately a little busy but it was really nice to work on ST2 and I am planning to keep sticking around. So just let me know if I should rebase the branch or anything else.

**Review by cognifloyd — APPROVED** [review]
Woohoo! Awesome. Thanks for working on this.

This needs 1 more approval, and @nzlosh needs to decide if it can land in 3.8 or if it needs to wait till after we finish cutting that release.

File: CHANGELOG.rst
Comment (amanda11) [inline CHANGELOG.rst:72]: This is in the changes made for stackstorm 3.8 - when should be in the new development section.

**Review by amanda11 — COMMENTED** [review]
Minor comment on changelog, and I'm wondering if its possible to add some unit-tests for the changes.

**Comment by cognifloyd** [comment]
I just went through a bunch of tests to see which tests deal somehow with publishing messages on the queue. In most cases, it is mocking or ignoring the message queue publishing.

Based on that, I think we could add some tests in [`st2common/tests/unit/test_transport.py`](https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_transport.py). The new test(s) can use the `QueueConsumer` class similar to how `TransportUtilsTestCase.test_publish_compression()` does it to test that messages get published to the right queue. Instead of using the dummy [Exchange+Queue+PoolPublisher](https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_transport.py#L71-L76), we would use:
- `st2common.transport.actionalias.ACTIONALIAS_XCHG`
- `st2common.transport.actionalias.get_queue()`
  I think all we need to use this safely (safe for parallel tests that touch RabbitMQ)
  is a test-specific `queue_name`, and then:
  `get_queue(name=queue_name, routing_key="#", exclusive=True, auto_delete=True)`
- `st2common.transport.actionalias.ActionAliasPublisher()`.

I don't see how we can test this without effectively testing kombu+RabbitMQ, since much of the added stuff creates the relevant queues and sets it up for publishing.

We could also add a db test similar to the `test_db_*.py` files that mocks `ActionAliasPublisher` and makes sure that `publish()` gets called when we create/update/delete (but not read) with [`ActionAliasDB`](https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/actionalias.py).

@amanda11 Does that sound like what you'd like to see? Would tests like this have value?

<hr/>

<details>
<summary>Here are my raw notes:</summary>
- uses liveaction and workflow publisher mocks from st2tests.mocks (see below)
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/services/test_workflow*.py
- mocks PoolPublisher.publish to ignore publishing
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/services/test_action.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_db.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_db_execution.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_db_liveaction.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_db_rule_enforcement.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_db_task.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_db_workflow.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_executions_util.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_param_utils.py
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_sensor_watcher.py
- checks that LiveActionPublisher.publish_state gets called and mocks PoolPublisher.publish to ignore CRUD publishing
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_action_db_utils.py
- tests st2common.transport.connection_retry_wrapper
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_connection_retry_wrapper.py
- tests st2common.transport.publishers.StatePublisherMixin
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_state_publisher.py
- tests st2common.stream.listener with mocked transport messages
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_stream_generator.py
- tests st2common.transport misc
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/unit/test_transport.py
    - `TransportUtilsTestCase.test_publish_compression`:
      tests `st2common.transport.publishers.PoolPublisher.publish()`
      with various `[messaging].compression` st2.conf settings
    - `TransportUtilsTestCase.test_get_ssl_kwargs`:
      tests `st2common.transport.utils._get_ssl_kwargs()`
- tests listening with SSL settings
  - https://github.com/StackStorm/st2/blob/master/st2common/tests/integration/test_rabbitmq_ssl_listener.py


mock publishers in:
- https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/mocks/execution.py
- https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/mocks/liveaction.py
- https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/mocks/workflow.py
</details>

**Comment by rush-skills** [comment]
@cognifloyd @ubaumann Is this still WIP?

**Comment by cognifloyd** [comment]
I would love to see this merged. I'm satisfied with this without the tests. But, @amanda11 mentioned adding tests, so I tried to figure out what tests we could add. @rush-skills wdyt? Does this need tests?

**Review by amanda11 — APPROVED** [review]
I'm happy to merge as is, without the extra tests. I'd rather it be in, than wait months for tests...

**Pull request merged**
